src/main/java/net/snowflake/ingest/streaming/internal/ParquetRowBuffer.java

```diff
@@ -169,15 +168,18 @@ private float addRow(
       RowBufferStats forkedStats = statsMap.get(columnName).forkEmpty();
       forkedStatsMap.put(columnName, forkedStats);
       ColumnMetadata column = parquetColumn.columnMetadata;
-      ParquetValueParser.ParquetBufferValue valueWithSize =
-          ParquetValueParser.parseColumnValueToParquet(
-              value,
-              column,
-              parquetColumn.type,
-              forkedStats,
-              defaultTimezone,
-              insertRowsCurrIndex,
-              clientBufferParameters.isEnableNewJsonParsingLogic());
+      ParquetBufferValue valueWithSize =
+          (clientBufferParameters.getIsIcebergMode()
+              ? IcebergParquetValueParser.parseColumnValueToParquet(
```

Comment:
any impact of the whole isEnableNewJsonParsingLogic business for iceberg parser?

Reply:
The `isEnableNewJsonParsingLogic` is used for semi structured data type, which will not impact Iceberg parsing.